### PR TITLE
Add PatchService method in service/helper.

### PR DESCRIPTION
**Pull request by prameshj**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR adds PatchService helper method to cloud-provider/service/helpers.go. Currently this is used by k/k service controller. But there are other codepaths in legacy-cloud-provider/gce and more that will benefit from this method and this is better place rather than utils within k/k. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go [hunk 1]
```diff
@@ -269,3 +271,78 @@ func TestHasLBFinalizer(t *testing.T) {
 		})
 	}
 }
+
+func TestPatch(t *testing.T) {
```
Comment (andrewsykim) [inline staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go]: nit: `TestPatchService`

File: staging/src/k8s.io/cloud-provider/service/helpers/helper.go [hunk 2]
```diff
@@ -17,10 +17,14 @@ limitations under the License.
...
 
 import (
+	"encoding/json"
 	"fmt"
 	"strings"
 
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
```
Comment (andrewsykim) [inline staging/src/k8s.io/cloud-provider/service/helpers/helper.go]: nit: corev1?

File: staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go [hunk 3]
```diff
@@ -269,3 +271,78 @@ func TestHasLBFinalizer(t *testing.T) {
...
+	if !reflect.DeepEqual(svcPatched, svcToPatch) {
+		t.Errorf("PatchStatus() = %+v, want %+v", svcPatched, svcToPatch)
+	}
+	// Explicitly validate if spec is unchanged from origin.
+	if !reflect.DeepEqual(svcPatched.Spec, svcOrigin.Spec) {
+		t.Errorf("Got spec = %+v, want %+v", svcPatched.Spec, svcOrigin.Spec)
+	}
+}
+
+func TestGetPatchBytes(t *testing.T) {
```
Comment (andrewsykim) [inline staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go]: nit: `Test_getPatchBytes`?

**Review by andrewsykim — COMMENTED** [review]
minor comments, lgtm otherwise :)

File: staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go (hunk 1 above)
Comment (prameshj) [reply staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go]: done

File: staging/src/k8s.io/cloud-provider/service/helpers/helper.go (hunk 2 above)
Comment (prameshj) [reply staging/src/k8s.io/cloud-provider/service/helpers/helper.go]: done

**Review by MrHohn — APPROVED** [review]
Thanks for adding this!
/lgtm

File: staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go (hunk 3 above)
Comment (prameshj) [reply staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go]: done

**Comment by prameshj** [comment]
Thanks for the review, @MrHohn @andrewsykim 
Pinging for approval.

**Comment by prameshj** [comment]
@andrewsykim could you help create a release in k8s.io/cloud-provider repo including this change? Thanks!

**Comment by andrewsykim** [comment]
@prameshj @k8s-publishing-bot should update the repo automatically at some point

**Pull request merged**